### DISCUSSION TITLE
How do I render gaps or nulls or breaks in the data

### DISCUSSION THREAD

**Discussion by jyasuna** [GSI-CS-CO/chart-fx#413] [Q&A]
I am starting with this data (just as an example)

Double[] x = new Double[]{1d,2d,**_3d_**,4d,5d};
Double[] y = new Double[]{1d,2d,**_null_**,4d,5d};

When converting the data to primitives for the DataSet the "null" becomes "Double.NaN" (I do the conversion)

I had hoped I would then see a simple line plot with a break in it, meaning a line for 1->2 and then a line from 4->5

But instead I see a "sawtooth" where line is drown from 2 down to the x-axis and back up again to 3

So, how do I handle "breaks" in my data? I just want that {2,null} to be considered in x-axis ranging but not drawn

**Comment by jyasuna** [comment]
So,  I tried something that almost worked

I set ErrorDataSetRenderer.setAllowNaNs(true)

Then I sent in the following data (converting nulls to Double.NaN of course)

new Integer[]{1,2,3,4,**_5_**,6,7,8,9,10},
new Double[]{1d,2d,3d,4d,**_null_**,6d,7d,8d,9d,10d},

neither 4->5 nor 5->6 segments were drawn (that's good),
however 3->4 segment also was not drawn (that's bad)

any ideas ?

I am trying to move all of the development away from JFreechart and instead using chart-fax ...


![Capture](https://user-images.githubusercontent.com/3989992/117035400-4410e680-acd2-11eb-8ffe-f24953ceea54.PNG)

**Reply by wirew0rm** [reply]
Hey, glad to see that you already found `setAllowNaNs()`. The reason this is not enabled by default is that before javafx11, NaNs were not handled propertly. Could you try to disable data reduction on the renderer? This is not the expected behavior, but if the error is there this would be a viable workaround for your current number of points and would narrow down the problem.

Hope this helps, Alex.

**Comment by jyasuna** [comment]
I tried **ErrorDataSetRenderer** with "**setPointReduction(false)** & **setAllowNaNs(true)**" and indeed this did work

I also tried using **ReducingLineRenderer** and that worked as well

**Reply by wirew0rm** [reply]
Ok ,thanks for confirming. I'll have to add some testcases to the point reduction algorithm which include NaNs.

**Reply by wirew0rm** [reply]
I did a fix for this in #414, thanks for reporting will probably be fixed in the next release.

**Comment by jyasuna** [comment]
Hello,

  Do you have an example where the data along one of the axes are Integers only?

I am trying to do this, and if I zoom in too tightly the grid lines are not placed properly?

 

I am sure the issue is on my end, which is why I am asking to see if you have an example like this?

 

Thx … jules

**Reply by RalphSteinhagen** [reply]
Hi, this shouldn't be an issue ... there are quite many examples in [samples](https://github.com/GSI-CS-CO/chart-fx/tree/master/chartfx-samples/src/main/java/de/gsi/chart/samples) where the x-values are integers.

You may check out the different samples also as shown [here](https://github.com/GSI-CS-CO/chart-fx#user-content-examples).
